### ISSUE TITLE
Regression rescue not firing.

### ISSUE THREAD

**Issue by CorbanR** [ansible/ansible#15715]
ISSUE TYPE

Bug Report
ANSIBLE VERSION

ansible: `2.0.2.0`
OS / ENVIRONMENT

`Ubuntu 14.04`

SUMMARY

After upgrading to Ansible 2.0.2.0 rescue is no longer triggering

STEPS TO REPRODUCE

Given the following playbook:

```
- name: include continuous.yml
  include: continuous.yml

#continuous.yml
- block:
    - name: Run migrations
      command: "./bin/rake db:migrate"
      args:
        chdir: "{{ app_base }}"
      environment: "{{ env_vars_merged }}"
      become_user: "{{ app_user }}"
      run_once: true
  rescue:
    - name: Send Slack notification on failure
      slack:
        token: "{{ slack_token }}"
        color: warning
        msg: "Migrations failed"
        channel: "{{ slack_channel }}"
        username: "{{ slack_username }}"
        icon_url: "{{ slack_icon_url }}"
      run_once: true
```

when the migrations run (and in this case fail) the rescue portion of the block is NOT triggered.

EXPECTED RESULTS

I would expect the rescue portion to be triggered

ACTUAL RESULTS

Rescue is not triggered.

**Comment by CorbanR** [comment]
After a little bit more research this is very similar to https://github.com/ansible/ansible/issues/14324. 

When will that patch be released?

**Comment by CorbanR** [comment]
There seems to be another regression.. when including a task and the included task has .. something like 

```
- hosts: all
  become: yes
  tasks:
    - name: some task
      command: "cat /home/some.txt"
```

you get 

`atal: [localhost]: FAILED! => {"failed": true, "reason": "no action detected in task. This often indicates a misspelled module name, or incorrect module path.\n\nThe error appears to have been in 'REDACTED': line 84, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- hosts: ALL\n  ^ here\n\n\nThe error appears to have been in 'REDACTED': line 84, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- hosts: all\n  ^ here\n"}`

removing the `- host` from the include in putting it in the `main.yml` it works with no problems.

**Comment by CorbanR** [comment]
Both issues I mentioned seem to only be a problem when using `include: playbook.yml`

**Comment by jimi-c** [comment]
@CorbanR the patch above will be included in 2.1. If you've verified that patch fixes your issue, we can close this as a dupe.

**Comment by einarc** [comment]
@jimi-c  It is not clear if the patch fixes the issue with doing the following:

`include: playbook.yml`
And then the playbook having a hosts entry:
`- hosts: all
  become: yes
  tasks:
    - name: some task
      command: "cat /home/some.txt"`

Will throw the error mentioned above:

> atal: [localhost]: FAILED! => {"failed": true, "reason": "no action detected in task. This often indicates a misspelled module name, or incorrect module path.\n\nThe error appears to have been in 'REDACTED': line 84, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- hosts: ALL\n ^ here\n\n\nThe error appears to have been in 'REDACTED': line 84, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- hosts: all\n ^ here\n"}

Should I file a separate bug for that?

FYI, upgrading to ansible 2.1 TODAY did not fix that issue, the issue is already reported here:
https://groups.google.com/forum/#!topic/ansible-project/BjycVAVlUpI
